lib/ansible/plugins/loader.py

```diff
@@ -324,6 +324,20 @@ def find_plugin(self, name, mod_type='', ignore_deprecated=False, check_aliases=
 
         return None
 
+    def find_plugin(self, name, mod_type='', ignore_deprecated=False, check_aliases=False):
+        ''' Find a plugin named name '''
+
+        # Import here to avoid circular import
+        from ansible.vars.reserved import is_reserved_name
+
+        plugin = self._find_plugin(name, mod_type=mod_type, ignore_deprecated=ignore_deprecated, check_aliases=check_aliases)
+        if plugin and self.package == 'ansible.modules' and is_reserved_name(name):
```

Comment:
Since this is limited to the `ansible.modules` package, is it able to catch any issues that would not be detected by a sanity test?

Reply:
The purpose of this PR is not to resolve a CI related problem or to resolve issues with module submissions.  It is however designed to solve issues our users face when they may have a file that is picked up as a module, that shadows the name of a keyword.  There are 2 example issues linked to this PR that showcase the issue.